### DISCUSSION TITLE
how to add xdebug?

### DISCUSSION THREAD

**Discussion by mnelson4** [leokhoa/laragon#391] [Q&A] [answered]
I just added PHP 8.1 to laragon, so now I need to enable xdebug for it. How is that done? (The old discussion item went away)

**Comment by Pen-y-Fan** [comment, marked as answer]
I wrote this blog [How to Set up VS Code to use PHP with Xdebug 3 on Windows](https://pen-y-fan.github.io/2021/08/03/How-to-Set-up-VS-Code-to-use-PHP-with-Xdebug-3-on-Windows/), which should help.

I have seen problems with automatically enabling xDebug in Laragon. If you have problems see the fix:

- [Xdebug can not be automatically enabled #332](https://github.com/leokhoa/laragon/issues/332)

If you use Nginx there is a conflict with port 9003, the workaround is to use the old post 9000. See:

- [Port conflict when using xDebug with Nginx #374](https://github.com/leokhoa/laragon/issues/374)

After it's been enabled once and the port doesn't conflict it works well and can be toggled on and off as required 🎉

**Comment by Pen-y-Fan** [comment]
I also have a snippet on xDebug on my blog: [How to update the PHP version in Laragon](https://pen-y-fan.github.io/2023/01/15/how-to-update-the-php-version-in-laragon/#xdebug)

**Comment by mnelson4** [comment]
Awesome thanks @Pen-y-Fan

**Comment by gamahachaa** [comment]
There is also a nice wizard on the xdebug website that will help you install it flawlessly by just analyzing your phpinfo(); output !
Easy and work seamlessly:

https://xdebug.org/wizard

**Reply by marcinlentner** [reply]
This is the way

**Comment by jovelluna12** [comment]
tried to set up xdebug through the wizard (link below), but the xdebug dll file it told me to download has been blocked, saying it is a potentially dangerous file. Is this file even safe? 

https://xdebug.org/wizard

**Reply by Pen-y-Fan** [reply]
The 'potentially dangerous file' message is a standard warning for any executable file download. As long as the file is downloaded from the official website, it is fine.

**Comment by slolo2000** [comment]
Hello,

I have read the wiki [How to add xdebug to Laragon](https://github.com/leokhoa/laragon/wiki/%5BTutorial%5D-How-to-add-Xdebug-to-Laragon)
> Instructions
> 
>     Download php_xdebug-3.4.1-8.3-nts-vs16-x86_64.dll
>     Move the downloaded file to L:\Laragon\bin\php\php-8.3.16-nts-Win32-vs16-x64\ext, and rename it to php_xdebug.dll
>     Update L:\Laragon\bin\php\php-8.3.16-nts-Win32-vs16-x64\php.ini to have the line:
>     zend_extension = xdebug
>     Restart the webserver

But when you read multiple discussions, it seems that this is not the good way to add xdebug ([see here](https://github.com/leokhoa/laragon/issues/332))

Is it the wiki of Laragon that is not up to date?

Is it really necessary to add [xdebug] section to the php.ini file in order to make it works?

**Reply by AWqxKAWERbXo** [reply]
Which way should be the best way instead?

**Reply by slolo2000** [reply]
Not the best way but the way it has to be correctly configured to work.

As I said, the Wiki give a way to do it (without mention to add [xdebug] section in php.ini file) and another one (https://github.com/leokhoa/laragon/issues/332) which tell you to add a [xdebug] section.

To be simple, do the user have to create the section himself like below into php.ini file?
```
[xdebug]
; these two lines are required for xDebug 3+
zend_extension = xdebug
xdebug.mode=coverage,debug,develop
; Laragon has a conflict with Nginx and xDebug using port 9003, set xDebug to use 9000.
xdebug.client_port=9000

```

**Reply by AWqxKAWERbXo** [reply]
@leokhoa in Laragon 8.6, you changed the default port from nginx. Right? So this information changig the xDebug Port isn't up to date anymore, is it?

**Reply by leokhoa** [reply]
@AWqxKAWERbXo : Thanks for asking. That's right. Laragon should work perfectly with XDebug without changing it's default port.

**Comment by pgkerr76** [comment]
I just started using laragon and had trouble getting xdebug to actually start working.
I installed php 8.5 after installing laragon, confirmed php.ini was loading from correct php version/directory and [xdebug] section at bottom of ini.
Confirmed correct xdebug version for 8.5 saved to correct ext directory.
It would not show in phpinfo(). **Nothing I tried worked until I added  laragon to path**

As a side note i hade tried to get zip enabled too. Naturally zip was already in php.ini. when using laragon menu to enable exstenions zip did not work either even though laragon was removing the " ;  " from the begining of the line. Again once I added laragon to path eveything started working
